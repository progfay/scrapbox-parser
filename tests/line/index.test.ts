/* global describe it expect */

describe('line', () => {
  it('Line that have multi node', () => {
    expect('[Link][Link]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'link',
            pathType: 'relative',
            href: 'Link',
            content: ''
          },
          {
            type: 'link',
            pathType: 'relative',
            href: 'Link',
            content: ''
          }
        ]
      }
    ])
  })

  it('Decoration line includes internal link', () => {
    expect('[* [Link]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'decoration',
            decos: ['*-1'],
            nodes: [
              {
                type: 'link',
                pathType: 'relative',
                href: 'Link',
                content: ''
              }
            ]
          }
        ]
      }
    ])
  })

  it('Decoration line includes internal link', () => {
    expect('[* [https://example.com example]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'decoration',
            decos: ['*-1'],
            nodes: [
              {
                type: 'link',
                pathType: 'absolute',
                href: 'https://example.com',
                content: 'example'
              }
            ]
          }
        ]
      }
    ])
  })

  it('Multi `]`', () => {
    expect('[* [Link]`code`[Link]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'decoration',
            decos: ['*-1'],
            nodes: [
              {
                type: 'plain',
                text: '[Link'
              }
            ]
          },
          {
            type: 'code',
            text: 'code'
          },
          {
            type: 'link',
            pathType: 'relative',
            href: 'Link',
            content: ''
          },
          {
            type: 'plain',
            text: ']'
          }
        ]
      }
    ])
  })
})
