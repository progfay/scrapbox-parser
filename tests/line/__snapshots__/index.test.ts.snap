// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line Decoration line includes internal link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "decos": Array [
          "*-1",
        ],
        "nodes": Array [
          Object {
            "content": "",
            "href": "Link",
            "pathType": "relative",
            "raw": "[Link]",
            "type": "link",
          },
        ],
        "raw": "[* [Link]]",
        "rawDecos": "*",
        "type": "decoration",
      },
    ],
    "type": "line",
  },
]
`;

exports[`line Decoration line includes internal link 2`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "decos": Array [
          "*-1",
        ],
        "nodes": Array [
          Object {
            "content": "example",
            "href": "https://example.com",
            "pathType": "absolute",
            "raw": "[https://example.com example]",
            "type": "link",
          },
        ],
        "raw": "[* [https://example.com example]]",
        "rawDecos": "*",
        "type": "decoration",
      },
    ],
    "type": "line",
  },
]
`;

exports[`line Line that have multi node 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "Link",
        "pathType": "relative",
        "raw": "[Link]",
        "type": "link",
      },
      Object {
        "content": "",
        "href": "Link",
        "pathType": "relative",
        "raw": "[Link]",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`line Multi \`]\` 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "decos": Array [
          "*-1",
        ],
        "nodes": Array [
          Object {
            "raw": "[Link",
            "text": "[Link",
            "type": "plain",
          },
        ],
        "raw": "[* [Link]",
        "rawDecos": "*",
        "type": "decoration",
      },
      Object {
        "raw": "\`code\`",
        "text": "code",
        "type": "code",
      },
      Object {
        "content": "",
        "href": "Link",
        "pathType": "relative",
        "raw": "[Link]",
        "type": "link",
      },
      Object {
        "raw": "]",
        "text": "]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;
